core/src/main/java/com/scalar/db/api/Result.java

```diff
@@ -37,7 +42,150 @@
+
+  /**
+   * Returns the FLOAT value for the specified name as a Java float type.
+   *
+   * <p>Note that, due to its signature, this method cannot return null. If the value is NULL, it
+   * will return 0.0. If this doesn't work for you, either call {@link #isNull(String)} before
+   * calling this method, or use {@link #getObject(String)} instead.
+   *
+   * @param name a name
+   * @return the FLOAT value for the specified name as a Java float type
+   */
+  float getFloat(String name);
+
+  /**
+   * Returns the DOUBLE value for the specified name as a Java double type.
+   *
+   * <p>Note that, due to its signature, this method cannot return null. If the value is NULL, it
+   * will return 0.0. If this doesn't work for you, either call {@link #isNull(String)} before
+   * calling this method, or use {@link #getObject(String)} instead.
+   *
+   * @param name a name
+   * @return the DOUBLE value for the specified name as a Java double type
+   */
+  double getDouble(String name);
+
+  /**
+   * Returns the TEXT value for the specified name as a Java String type.
+   *
+   * @param name a name
+   * @return the TEXT value for the specified name as a Java String type. If the value is NULL, null
+   */
+  @Nullable
+  String getText(String name);
+
+  /**
+   * Returns the BLOB value for the specified name as a Java ByteBuffer type.
+   *
+   * @param name a name
+   * @return the BLOB value for the specified name as a Java ByteBuffer type. If the value is NULL,
+   *     null
+   */
+  @Nullable
+  default ByteBuffer getBlob(String name) {
+    return getBlobAsByteBuffer(name);
+  }
+
+  /**
+   * Returns the BLOB value for the specified name as a Java ByteBuffer type.
+   *
+   * @param name a name
+   * @return the BLOB value for the specified name as a Java ByteBuffer type. If the value is NULL,
+   *     null
+   */
+  @Nullable
+  ByteBuffer getBlobAsByteBuffer(String name);
+
+  /**
+   * Returns the BLOB value for the specified name as a Java byte array type.
+   *
+   * @param name a name
+   * @return the BLOB value for the specified name as a Java byte array type. If the value is NULL,
+   *     null
+   */
+  @Nullable
+  byte[] getBlobAsBytes(String name);
+
+  /**
+   * Returns the value for the specified name as a Java Object type.
+   *
+   * <p>If the columns is a BOOLEAN type, it returns a {@code Boolean} object. If the columns is an
+   * INT type, it returns an {@code Integer} object. If the columns is a BIGINT type, it returns a
+   * {@code LONG} object. If the columns is a FLOAT type, it returns a {@code FLOAT} object. If the
+   * columns is a DOUBLE type, it returns a {@code DOUBLE} object. If the columns is a TEXT type, it
+   * returns a {@code String} object. If the columns is a BLOB type, it returns a {@code ByteBuffer}
+   * object.
+   *
+   * @param name a name
+   * @return the value for the specified name as a Java Object type. If the value is NULL, null
+   */
+  @Nullable
+  Object getObject(String name);
```

Comment:
Also, I added several method to get a value from the result. After this change, users no longer need to treat the `Value` class directly. So we can completely hide this class from users. I'm going to do that in another PR.